JYP-iOS/JYP-iOS/Sources/Presenter/NavigationBar/NavigationBarViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  NavigationBarViewController.swift
+//  JYP-iOS
+//
+//  Created by 송영모 on 2022/07/30.
+//  Copyright © 2022 JYP-iOS. All rights reserved.
+//
+
+import UIKit
+
+class NavigationBar: UIView {
+    var backButton = UIButton()
+    var title = UILabel()
+    var subTitle = UILabel()
+}
+
+protocol BaseNavigationBarViewControllerProtocol: AnyObject {
+    var statusBar: UIView { get }
+    var navigaionBar: NavigationBar { get }
+    var contentView: UIView { get }
+    
+    func setupNavigationBar()
+    func setNavigationBarHidden(_ hidden: Bool)
+    func setNavigationBarBackButtonHidden(_ hidden: Bool)
+    func setNavigationBarBackButtonTitleColor(_ color: UIColor?)
+    func setNavigationBarTitleText(_ text: String?)
+    func setNavigationBarTitleFont(_ font: UIFont?)
+    func setNavigationBarTitleTextColor(_ color: UIColor?)
+    func setNavigationBarSubTitleText(_ text: String?)
+    func setNavigationBarSubTitleFont(_ font: UIFont?)
+    func setNavigationBarSubTitleTextColor(_ color: UIColor?)
+}
+
+class NavigationBarViewController: BaseViewController, BaseNavigationBarViewControllerProtocol {
+    // MARK: - UI Components
+    
+    var statusBar = UIView()
+    var navigaionBar = NavigationBar()
+    var contentView = UIView()
+    
+    // MARK: - Properties
+    
+    private let titleFont = JYPIOSFontFamily.Pretendard.medium.font(size: 16)
+    private let titleTextColor = JYPIOSAsset.textB75.color
+    private let subTitleFont = JYPIOSFontFamily.Pretendard.regular.font(size: 18)
+    private let subTitleTextColor = JYPIOSAsset.tagGrey200.color
+    
+    // MARK: - Life Cycle Methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupNavigationBar()
+    }
+    
+    // MARK: - Setup Methods
+    
+    override func setupProperty() {
+        super.setupProperty()
+        
+        navigationController?.setNavigationBarHidden(true, animated: false)
```

Comment:
```suggestion
        navigationController?.setNavigationBarHidden(true, animated: false)
        navigationController?.interactivePopGestureRecognizer?.delegate = nil
```

navigationBar가 hidden일 때는 스와이프 백 제스처가 동작하지 않아 아래와 같은 처리가 필요하다고 생각합니다

Reply:
[공식 문서](https://developer.apple.com/documentation/uikit/uinavigationcontroller/1621847-interactivepopgesturerecognizer)
[이슈 예시](https://borabong.tistory.com/14)

학습 완, 반영 완료했습니다 ! 감사합니당